packages/button/src/LionButton.js

```diff
@@ -140,32 +145,42 @@ export class LionButton extends DelegateMixin(SlotMixin(LionLitElement)) {
 
   connectedCallback() {
     super.connectedCallback();
-    this.__setupKeydownDelegation();
+    this.__setupDelegation();
   }
 
   disconnectedCallback() {
     super.disconnectedCallback();
-    this.__teardownKeydownDelegation();
+    this.__teardownDelegation();
   }
 
   __clickDelegationHandler(e) {
     e.stopPropagation(); // prevent click on the fake element and cause click on the native button
     this.$$slot('_button').click();
   }
 
-  __setupKeydownDelegation() {
+  __setupDelegation() {
     this.addEventListener('keydown', this.__keydownDelegationHandler);
+    this.addEventListener('keyup', this.__keyupDelegationHandler);
   }
 
-  __teardownKeydownDelegation() {
+  __teardownDelegation() {
     this.removeEventListener('keydown', this.__keydownDelegationHandler);
+    this.removeEventListener('keyup', this.__keyupDelegationHandler);
   }
 
   __keydownDelegationHandler(e) {
+    if (e.keyCode === 32 /* space */ || e.keyCode === 13 /* enter */) {
+      e.preventDefault();
+      this.shadowRoot.querySelector('.btn').classList.add('active');
+    }
+  }
+
+  __keyupDelegationHandler(e) {
     // Makes the real button the trigger in forms (will submit form, as opposed to paper-button)
     // and make click handlers on button work on space and enter
     if (e.keyCode === 32 /* space */ || e.keyCode === 13 /* enter */) {
       e.preventDefault();
+      this.shadowRoot.querySelector('.btn').classList.remove('active');
```

Comment:
One of the most interesting findings during my work on this is that we can have this simple logic of removing `.active` from `.btn` without adding extra logic for corner cases when `keyup` does not happen on `lion-button`. e.g. if someone keeps pressing Enter (keydown) and then clicks somewhere, so this logic of removal will not be run and the `lion-button` will be stuck in the active state. Why? Because this is exactly how native button works with `:active`, it also remembers the active state in this situation.

So in both native and this custom implementation this can be undone only after another focus on the button and another sequence of keydown/keyup. So we are aligned with the platform here. Surprising, but yeah :) Platform is sometimes pretty surprising.

(I didn't do extensive crossbrowser testing of how native button behaves though, but I also know that it often behaves differently on different browsers, so consistency is almost impossible to achieve anyway)

Reply:
Interesting. Almost sounds like a feature. Although would be interesting to see what other browsers do, since it wouldn't be too hard for us to fix this, I would say, platform bug

Reply:
Let's please keep it for the other MR, because fixing browser inconsistencies is an issue on its own which we are not even sure about.